js/stores/data-store.js

```diff
@@ -64,28 +64,42 @@ class DataStore extends StoreBoilerplate{
     //register the action callbacks
     dispatcher.register((action) => {
       let overridden = false;
+      let actionState = _this.state;
+      let continueIfUpdatingState = function(method){
+        actionState = method(action, actionState, _this);
+        return !!actionState;
+      }
 
       if(_actionHandlers.hasOwnProperty(action.actionType)){
         _actionHandlers[action.actionType]
           .prePatches
-          .forEach(method => _this.state = method(action, _this.state));
+          .every(continueIfUpdatingState);
+
+        // If the action is forcing a change not to result in an emit, return.
+        if (actionState === null) { return; }
```

Comment:
No, absolutely.  I'm sure there's _something_ we can do to improve this, but it didn't quite jump out at me, eh.

Anyhow, we're going to prevent the 'next-step' functions from firing if the state becomes null.